extensions/machine-learning-services/src/views/models/modelManagementController.ts

```diff
@@ -51,11 +51,16 @@ export class ModelManagementController extends ControllerBase {
 	 * @param apiWrapper apiWrapper
 	 * @param root root folder path
 	 */
-	public async registerModel(parent?: ModelViewBase, controller?: ModelManagementController, apiWrapper?: ApiWrapper, root?: string): Promise<ModelViewBase> {
+	public async registerModel(importTable: DatabaseTable | undefined, parent?: ModelViewBase, controller?: ModelManagementController, apiWrapper?: ApiWrapper, root?: string): Promise<ModelViewBase> {
```

Comment:
Another learning moment for me: what's the difference between `importTable: DatabaseTable | undefined` and `importTable?: DatabaseTable`?

Reply:
if the type can be undefiled you need to specify that in the type. ? means it's optional parameter

---
In reply to: [411748081](https://github.com/microsoft/azuredatastudio/pull/10071#discussion_r411748081) [](ancestors = 411748081)

Reply:
I see... thought optional parameters got set to `undefined` automatically, so that became inherent in their typing